### PR TITLE
use_local_updates in optimizer

### PR DESCRIPTION
**Pull request by justheuristic**
[this PR fixes some edge case bugs introduced since 1.1.0, those bugs do not affect pip version]
- fix use_local_updates in Optimizer, add test cases to ensure that it works
- ensure that grad_compression is always used

### CODE REVIEW

**Review by artek0chumak — APPROVED** [review]

File: tests/test_optimizer.py [hunk 1]
```diff
@@ -391,6 +430,6 @@ def run_trainer(batch_size: int, batch_time: float, client_mode: bool):
     assert 4 / 0.3 * 0.8 <= optimizer.tracker.performance_ema.samples_per_second <= 4 / 0.3 * 1.2
 
     assert not optimizer.state_averager.is_alive()
-    assert not optimizer.grad_averager.is_alive()
     assert not optimizer.tracker.is_alive()
+    assert optimizer.grad_averager is None if use_local_updates else (not optimizer.grad_averager.is_alive())
```
Comment (mryab) [inline tests/test_optimizer.py]: Better to split the conditional in multiple lines:

Comment (mryab) [inline tests/test_optimizer.py]:
```suggestion
    if use_local_updates:
        assert optimizer.grad_averager is None
    else:
        assert not optimizer.grad_averager.is_alive()
```

File: tests/test_optimizer.py [hunk 2]
```diff
@@ -297,12 +297,49 @@ def run_worker(index: int, batch_size: int, period: float, **kwargs):
...
+        (False, True, True, True, True),
+        (False, True, True, False, True),
+        (True, False, False, False, False),
+        (
+            True,
+            True,
+            False,
+            False,
+            False,
+        ),
```
Comment (mryab) [inline tests/test_optimizer.py:316]: To make formatting more consistent, you can surround the entire block of cases above with `# fmt: off` and `# fmt: on`

**Review by mryab — APPROVED** [review]

File: tests/test_optimizer.py (hunk 2 above)
Comment (justheuristic) [reply tests/test_optimizer.py]: good point, thx
